### PR TITLE
Added isWebhookMessage() and docs to (Guild)MessageReceivedEvent

### PR DESCRIPTION
**Pull request by kantenkugel**
Need opinion on Docs:
- Should I keep the `@see` stuff?
- Should I say "received Message" or "sent Message" throughout the docs?
- Should I link other classes more frequently? (currently only when needed and not in return statement)
- Should I horizontally align the `@return` and `@see` statements?

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java [hunk 1]
```diff
@@ -21,8 +21,10 @@
 
 /**
  * <b><u>MessageReceivedEvent</u></b><br>
- * Fired if a Message is sent in a {@link net.dv8tion.jda.core.entities.MessageChannel MessageChannel}.<br>
- * <br>
+ * Fired if a Message is sent in a {@link net.dv8tion.jda.core.entities.MessageChannel MessageChannel}.
+ * <br>This includes both {@link net.dv8tion.jda.core.entities.TextChannel TextChannel}
+ * and {@link net.dv8tion.jda.core.entities.PrivateChannel PrivateChannel}!
+ * <p>
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java]: This also includes Groups!

File: src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java [hunk 2]
```diff
@@ -21,8 +21,10 @@
 
 /**
  * <b><u>MessageReceivedEvent</u></b><br>
- * Fired if a Message is sent in a {@link net.dv8tion.jda.core.entities.MessageChannel MessageChannel}.<br>
- * <br>
+ * Fired if a Message is sent in a {@link net.dv8tion.jda.core.entities.MessageChannel MessageChannel}.
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java]: Due to the changes coming in the Message rewrite, this should probably be "if a Message is received..."

File: src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java [hunk 3]
```diff
@@ -35,38 +37,109 @@ public MessageReceivedEvent(JDA api, long responseNumber, Message message)
...
+     * @see net.dv8tion.jda.core.entities.User#isFake()
+     */
     public User getAuthor()
     {
         return message.getAuthor();
     }
 
+    /**
+     * Returns the Author of the Message sent as {@link net.dv8tion.jda.core.entities.Member Member} object.
+     * <br>This will be {@code null} in case of Message being sent in a {@link net.dv8tion.jda.core.entities.PrivateChannel PrivateChannel}
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java]: Or Group!

File: src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java (hunk 2 above)
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java]: Due to the changes coming in the Message rewrite, this should probably be "if a Message is received..."

File: src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java [hunk 4]
```diff
@@ -21,8 +21,10 @@
 
 /**
  * <b><u>MessageReceivedEvent</u></b><br>
- * Fired if a Message is sent in a {@link net.dv8tion.jda.core.entities.MessageChannel MessageChannel}.<br>
- * <br>
+ * Fired if a Message is sent in a {@link net.dv8tion.jda.core.entities.MessageChannel MessageChannel}.
+ * <br>This includes both {@link net.dv8tion.jda.core.entities.TextChannel TextChannel}
+ * and {@link net.dv8tion.jda.core.entities.PrivateChannel PrivateChannel}!
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java]: This also includes Groups!

File: src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java [hunk 5]
```diff
@@ -35,38 +37,109 @@ public MessageReceivedEvent(JDA api, long responseNumber, Message message)
...
+     */
     public User getAuthor()
     {
         return message.getAuthor();
     }
 
+    /**
+     * Returns the Author of the Message sent as {@link net.dv8tion.jda.core.entities.Member Member} object.
+     * <br>This will be {@code null} in case of Message being sent in a {@link net.dv8tion.jda.core.entities.PrivateChannel PrivateChannel}
+     * or {@link #isWebhookMessage() isWebhookMessage()} returning {@code true}.
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java]: Or Group!

File: src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java (hunk 2 above)
Comment (kantenkugel) [reply src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java]: Received vs sent?

**Comment by kantenkugel** [comment]
Will change the docs where group is missing and think about what dv8 meant with first one

File: src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java (hunk 2 above)
Comment (DV8FromTheWorld) [reply src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java]: Considering the event is called MessageReceived, should probably keep that in line.

Additionally, we are bringing to JDA a slight separation between the constructs that represent messages that are Sent vs those that are Received.

File: src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java [hunk 6]
```diff
@@ -35,38 +37,109 @@ public MessageReceivedEvent(JDA api, long responseNumber, Message message)
         this.message = message;
     }
 
+    /**
+     * Returns the received {@link net.dv8tion.jda.core.entities.Message Message} object.
+     * <br>The Message will be of type {@link net.dv8tion.jda.core.entities.impl.ReceivedMessage ReceivedMessage}.
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java]: Considering the impl package is not included in the documentation we cannot link to it here

Comment (kantenkugel) [reply src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java]: Oh yea... Should i just remove that completely?

File: src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java (hunk 2 above)
Comment (kantenkugel) [reply src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java]: Yea gonna rename every "sent" occurence to "received"

**Pull request merged**